Userland/Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -598,13 +598,22 @@ void NodeWithStyle::apply_style(const CSS::StyleProperties& computed_style)
     if (clear.has_value())
         computed_values.set_clear(clear.value());
 
-    auto overflow_x = computed_style.overflow_x();
-    if (overflow_x.has_value())
-        computed_values.set_overflow_x(overflow_x.value());
-
-    auto overflow_y = computed_style.overflow_y();
-    if (overflow_y.has_value())
-        computed_values.set_overflow_y(overflow_y.value());
+    auto overflow_x = computed_style.overflow_x().value_or(CSS::Overflow::Visible);
+    auto overflow_y = computed_style.overflow_y().value_or(CSS::Overflow::Visible);
+    // https://www.w3.org/TR/css-overflow-3/#overflow-control
+    // The visible/clip values of overflow compute to auto/hidden (respectively) if one of overflow-x or overflow-y is neither visible nor clip.
+    if (!AK::first_is_one_of(overflow_x, CSS::Overflow::Visible, CSS::Overflow::Clip) || !AK::first_is_one_of(overflow_y, CSS::Overflow::Visible, CSS::Overflow::Clip)) {
+        if (overflow_x == CSS::Overflow::Visible)
+            overflow_x = CSS::Overflow::Auto;
+        if (overflow_x == CSS::Overflow::Clip)
+            overflow_x = CSS::Overflow::Hidden;
+        if (overflow_y == CSS::Overflow::Visible)
+            overflow_y = CSS::Overflow::Auto;
+        if (overflow_y == CSS::Overflow::Clip)
+            overflow_y = CSS::Overflow::Hidden;
+    }
+    computed_values.set_overflow_x(overflow_x);
+    computed_values.set_overflow_y(overflow_y);
```

Comment:
The spec says "compute", should this be done in `StyleComputer` instead? Should this be observable via `getComputedStyle`? A test for that would be good :)

Reply:
yes, turned out it should be resolved in StyleComputer indeed. I moved it there and added another test :)